### PR TITLE
fix(parser): merge base property schemas in allOf collision to preserve array type

### PR DESCRIPTION
**Pull request by jsklan**
## Description

Fixes a bug where `allOf` used to narrow an array property's `items` type—without redeclaring `type: array`—produced `list<unknown>` instead of the correctly typed `list<Item>`.

**Failing pattern:**

```yaml
ItemList:
  allOf:
    - $ref: "#/components/schemas/PaginatedList"   # has results: { type: array, items: {} }
    - properties:
        results:
          items:
            $ref: "#/components/schemas/Item"       # only specifies items, no type: array
```

## Spec context

Per the OpenAPI/JSON Schema spec, `allOf` is not a merge—it means the instance must independently validate against every subschema. Code generators approximate this as a merge to produce a single type. In this case, a valid instance's `results` must be an array (subschema 1) whose items match `Item` (subschema 2). Both hold simultaneously, so the generated type should be `list<Item>`.

The fix uses `{ ...basePropSchema, ...overridePropSchema }` to carry forward base fields like `type: array` when the inline override omits them, with override keys always winning on conflicts.

## Spec compliance summary

| Pattern | Spec says | PR produces | Correct? |
|---------|-----------|-------------|----------|
| Narrowing items without type | Both hold; array + narrowed items | `{ type: array, items: SideDish }` | ✅ Yes |
| Metadata propagation (readOnly, description) | Annotations apply from all subschemas | Base annotations carry forward | ✅ Correct |
| Colliding annotations (description) | Implementation-defined | Override wins, base propagates if absent | ✅ Reasonable |
| Overlapping sub-properties on same key | Deep intersection of both property sets | Not affected by PR (pre-existing gap) | ⚠️ Pre-existing |
| Conflicting validation keywords (enum) | Intersection of valid sets | Override replaces base (lossy) | ❌ Pre-existing |

The last two rows are pre-existing limitations of the shallow-merge approach that this PR neither introduces nor worsens.

## Changes Made

- **Legacy importer (`convertObject.ts`):** Before converting an inline `allOf` element's properties, merge base property schemas from referenced `allOf` parents using `{ ...basePropSchema, ...overridePropSchema }`. Creates a new `mergedProperties` record and a shallow copy of `allOfElement` to avoid mutating the parsed OpenAPI document.
- **V3 importer (`ObjectSchemaConverter.ts`):** Same approach — collect properties from resolved referenced `allOf` parents, merge into inline override schemas before conversion.
- **Updated ETE snapshots:** Properties like `data` and `links` now include `docs` and `access: read-only` metadata inherited from base schemas via `allOf` — this is correct per the spec (annotations apply from all subschemas).
- **Added test fixtures (`allof-array-items-narrowing`):** Covers items narrowing without `type: array`, explicit `type: array` override, and string property narrowing via `allOf`.

## Known limitations (not introduced by this PR)

- **Shallow parent property collection:** Only direct properties of referenced `allOf` parents are considered, not properties inherited through nested `allOf` chains.
- **`$ref` base properties skipped:** Merge is skipped when the base property is a `$ref` (to avoid flattening named types).
- **First-vs-last parent inconsistency:** When multiple `$ref` parents define the same property, the legacy importer uses the first match while the V3 importer uses the last.

## Testing

- [x] `TEST_FIXTURE=allof-array-items-narrowing pnpm --filter @fern-api/v3-importer-tests test`
- [x] `TEST_FIXTURE=allof-oneof-anyof-edge-cases pnpm --filter @fern-api/openapi-ir-to-fern-tests test`
- [x] `pnpm --filter @fern-api/ete-tests exec vitest run src/tests/write-definition/writeDefinition.test.ts`
- [x] `pnpm run check` (biome lint)

### CODE REVIEW

File: packages/cli/api-importers/openapi/openapi-ir-parser/src/schema/convertObject.ts [hunk 1]
```diff
@@ -180,11 +180,39 @@ export function convertObject({
...
+                    let merged: OpenAPIV3.ReferenceObject | OpenAPIV3.SchemaObject = overridePropSchema;
+                    if (!isReferenceObject(overridePropSchema)) {
+                        for (const otherAllOfElement of allOf) {
+                            if (otherAllOfElement === allOfElement || !isReferenceObject(otherAllOfElement)) {
+                                continue;
+                            }
+                            const resolvedParent = context.resolveSchemaReference(otherAllOfElement);
+                            const basePropSchema = resolvedParent.properties?.[key];
+                            if (basePropSchema != null && !isReferenceObject(basePropSchema)) {
+                                merged = { ...basePropSchema, ...overridePropSchema };
```
[the remark this answers is not in the document]
Comment (jsklan) [reply packages/cli/api-importers/openapi/openapi-ir-parser/src/schema/convertObject.ts:201]: After chatting with Devin, I think this is per-spec behavior, so we don't need to modify it.

Details:

The snapshot diffs for the trains API (bookings.yml, stations.yml, trips.yml) show docs and access: read-only appearing on properties that didn't have them before. This is a side effect of the { ...basePropSchema, ...overridePropSchema } merge — it carries forward all base fields, not just structural ones like type and items.

Concrete example: Wrapper-Collection in the trains spec defines:

```
Wrapper-Collection:
  properties:
    data:
      description: The wrapper for a collection is an array of objects.
      type: array
      items:
        type: object
    links:
      description: A set of hypermedia links which serve as controls for the client.
      type: object
      readOnly: true
```

The GetBookingsResponse uses allOf with inline overrides that only specify data: { type: array, items: $ref Booking } and links: { allOf: [...] } — neither redeclares description or readOnly. After the merge, those base metadata fields propagate through because the override doesn't shadow them.

This is correct per the OpenAPI spec — allOf means the instance must validate against every subschema simultaneously, so annotations from all subschemas apply. The new output (docs: "The wrapper for a collection...", access: read-only on links) is more faithful to what the spec actually says.

**Review by Swimburger — APPROVED** [review]
